### PR TITLE
Janus: Send context via http requests

### PR DESCRIPTION
**Pull request by tomaszdurka**
Currently we send Context when connecting via websockets to Janus service (we do that on client-side https://github.com/cargomedia/cm/blob/2743980c9a58805a7c5df773cea5d7786647c4f6/library/CM/App.js#L246).
Let's do the same when sending plain http requests. 

--------------------------------------------
We send janus http requests using `CM_Janus_HttpApiClient#_request`. Let's add context parameter here.
Context can contain data about stream (reason for being stopped) or any other valid data.

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@fvovan please take a look

File: library/CM/Log/ContextFormatter/Cargomedia.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        ];
+        return array_merge($hash, $this->getContext($context));
+    }
+
+    /**
+     * @param CM_Log_Context $context
+     * @return array
+     */
+    public function getContext(CM_Log_Context $context) {
+        $hash = [];
```
Comment (fvovan) [inline library/CM/Log/ContextFormatter/Cargomedia.php]: I would rather name it as `$result`. `$hash` is too general

**Comment by fvovan** [comment]
`request` key should be actually `httpRequest`. it was fixed in recent PR
may it's good idea to prefix methods with `format` rather than `get`?

**Comment by tomaszdurka** [comment]
- Yes, I did the merge of upstream after my changes which resulted in mistakes.
- I am always a but sceptic about calling Formatters method `format*`. Mhmm.

**Comment by fvovan** [comment]
>I am always a but sceptic about calling Formatters method format*. Mhmm.

Agree but not sure if it's worse that calling non getters with `get`.

**Comment by tomaszdurka** [comment]
@fvovan please review

Please ignore `get*` namings for the moment (just read your messages, agree).

**Comment by tomaszdurka** [comment]
@njam any thoughts about the Cargomedia formatter?

**Comment by njam** [comment]
As long as it's in line with the [guidelines](https://github.com/cargomedia/guides/tree/master/logging) :)

Quickly looking at it I can see that "referrer" should be "referer", and "user_agent" should be "useragent".
Please check everything again.

File: library/CM/Janus/HttpApiClient.php [hunk 2]
```diff
@@ -4,12 +4,16 @@ class CM_Janus_HttpApiClient {
 
     /** @var \GuzzleHttp\Client */
     protected $_httpClient;
+    
+    /** @var CM_Log_ContextFormatter_Interface */
+    protected $_contextFormatter;
 
     /**
      * @param \GuzzleHttp\Client $httpClient
```
Comment (fvovan) [inline library/CM/Janus/HttpApiClient.php]: please adjust the doc

File: library/CM/Log/ContextFormatter/Cargomedia.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+class CM_Log_ContextFormatter_Cargomedia implements CM_Log_ContextFormatter_Interface {
+    
+    /** @var string */
+    protected $_appName;
+
+    /**
+     * @param string $appName
+     */
+    public function __construct($appName) {
```
Comment (fvovan) [inline library/CM/Log/ContextFormatter/Cargomedia.php]: is casting needed here?

**Comment by fvovan** [comment]
The rest looks good except mentioned by Reto wrong keys, `get..` and `$hash`

**Comment by tomaszdurka** [comment]
@fvovan please check

File: tests/library/CM/Http/Request/AbstractTest.php [hunk 4]
```diff
@@ -296,6 +296,10 @@ public function testGetTimeZoneFromCookie() {
         $this->assertNull($timeZone);
     }
 
+    public function testFactoryFromGlobals() {
```
Comment (fvovan) [inline tests/library/CM/Http/Request/AbstractTest.php]: is it needed?

File: tests/library/CM/Log/ContextFormatter/CargomediaTest.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
+<?php
+
```
Comment (fvovan) [inline tests/library/CM/Log/ContextFormatter/CargomediaTest.php]: please delete one line

**Comment by fvovan** [comment]
the rest lgtm

**Pull request merged**